plugins/hydra_orion_sweeper/hydra_plugins/hydra_orion_sweeper/_impl.py

```diff
@@ -0,0 +1,347 @@
+            storage=self.storage_config,
+            branching=self.orion_config.branching,
+            max_idle_time=None,
+            heartbeat=None,
+            working_dir=None,
+            debug=self.orion_config.debug,
+            executor=None,
+        )
+
+    def sweep(self, arguments: List[str]) -> None:
+        """Execute the optimization process"""
+
+        assert self.config is not None
+        assert self.launcher is not None
+        assert self.job_idx is not None
+
+        log.debug("Starting new experiment")
+        self.client = self.new_experiment(arguments)
+
+        with clientctx(self.client):
+            try:
+                self.optimize(self.client)
+            except Exception as e:
+                self.release_all()
+                raise e
+
+    def release_all(self) -> None:
+        """Make sure not trials remain reserved"""
+        for trial in self.pending_trials:
+            try:
+                self.client.release(trial, status="interrupted")
+            except AlreadyReleased:
+                pass
+
+    def optimize(self, client: ExperimentClient) -> None:
+        """Run the hyperparameter search in batches"""
+        failures = []
+
+        while not self.client.is_done:
+            trials = self.sample_trials()
+
+            if len(trials) == 0 and not self.client.is_done and not self.pending_trials:
+                raise RuntimeError()
+
+            returns = self.execute_trials(trials)
+
+            self.observe_results(trials, returns, failures)
+
+            if len(failures) > self.worker_config.max_broken:
+                # make the `Future` raise the exception it received
+                try:
+                    failures[-1].return_value
+                except Exception as e:
+                    raise BrokenExperiment("Max broken trials reached, stopping") from e
+
+        self.show_results()
+
+    def sample_trials(self) -> List[Trial]:
+        """Sample a new batch of trials"""
+
+        trials = self.suggest_trials(self.n_workers())
+        log.debug("Suggest %d new trials", len(trials))
+        self.pending_trials.update(set(trials))
+        return trials
+
+    def execute_trials(self, trials: List[Trial]) -> Sequence[JobReturn]:
+        """Execture the given batch of trials"""
+
+        overrides = list(as_overrides(t, self.arguments) for t in trials)
+        self.validate_batch_is_legal(overrides)
+
+        returns = self.launcher.launch(overrides, initial_job_idx=self.job_idx)
+        self.job_idx += len(returns)
+        return returns
+
+    def observe_results(
+        self,
+        trials: List[Trial],
+        returns: Sequence[JobReturn],
+        failures: Sequence[JobReturn],
+    ):
+        """Record the result of each trials"""
+        for trial, result in zip(trials, returns):
+            self.pending_trials.remove(trial)
+
+            if result.status == utils.JobStatus.COMPLETED:
+                value = result.return_value
```

Comment:
It would be great to verify whether `value` is a list, in which case it would already represent the dicts results for Oríon, otherwise we build the dict as you do below. Some users return more than just the objective.

Reply:
done